okhttp/build.gradle

```diff
@@ -1,9 +1,22 @@
 apply plugin: 'me.champeau.gradle.japicmp'
 
 jar {
-  manifest {
-    attributes('Automatic-Module-Name': 'okhttp3')
-  }
+  // modify these lines for MANIFEST.MF properties or for specific bnd instructions
+  // internal package needs to be exported too because interceptors use internal packages
+  bnd '''
+  Export-Package: \
+  okhttp3,\
+  okhttp3.internal.*
+  Import-Package: \
+  android.*;resolution:=optional,\
+  dalvik.system;resolution:=optional,\
+  org.conscrypt;resolution:=optional,\
+  org.bouncycastle.*;resolution:=optional,\
+  org.openjsse.*;resolution:=optional,\
```

Comment:
If we introduce new dependencies, is there a test that’ll fail to remind us to update this?

Reply:
If new dependencies are introduced you have to decide on a case by case basis if it's an optional dependency or not. If you forget to add the library in the list the worst case will be that a OSGi user of that library has to put that library on the classpath to ensure OSGi can resolve all imports. The resolution optional attribute basically means that those packages are not required at runtime.

Reply:
maybe a move the comment about optional (ex why not `Import-Package:  *`) here?

`Platform` or similar might need an update to mention when adding optional library support, to not forget to update this and wherever the test cases are! It could also be useful to use those paths in the OSGi test

Reply:
If we add a simple "*" here all the android/ssl packages would be required on the classpath for the bundle to resolve therefore I explicitly define the packages which are not required in every case at runtime.

Hmm... maybe we should add the comment to the build.gradle dependency block instead of the platform class because the dependency block must be touched to add new dependencies. I kind of dislike adding comments to code which are build related.

Reply:
I suppose I meant whatever the package-info equiv of kotlin. I suspect any new member of this directory would possibly taint here. https://github.com/square/okhttp/tree/master/okhttp/src/main/kotlin/okhttp3/internal/platform

Not too fussed where the comment goes, it can be in the CONTRIBUTING file.

thx for the clarification on not '*' that part could be a comment inline here in the gradle.

Reply:
Added documentation to the CONTRIBUTING file